internal/pkg/interp/interpreter.go

```diff
@@ -0,0 +1,425 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package interp defines an analyzer that performs taint analysis on
+// functions by interpreting their ssa code.
+package interp
```

Comment:
Are there unit tests specifically testing this?

Reply:
This is a good question. The answer is no.

The code in `internal/pkg/interp` is tested via the "end-to-end" tests under `internal/pkg/levee`.

The main reason I didn't write unit tests is that for the purposes of "trying out" this new approach I felt like the end-to-end tests were sufficient. I do think this new approach is preferable to the current approach, and it should probably have unit tests before (and if) it is merged.

The other reason why there are currently no unit tests is that I am not sure what those tests should look like. Ideally, I would like to test that processing each type of instruction has the intended effect, e.g. updating the `state`, appending a `result`, etc. I think this should be doable using `analysistest` and `want` comments, at least to some extent, but I am not sure what the cost/benefit would be, and how strongly the tests would have to be coupled to the implementation.